### DISCUSSION TITLE
Problem create queue "Type: Default for virtual host

### DISCUSSION THREAD

**Discussion by ko2024** [rabbitmq/rabbitmq-server#11359] [Other]
When trying to create a queue in rabbit version 3.13.2 using the type Default for virtual host mode, the error occurs:
in version 3.11.10 it works normally. Do you know if there are any particularities in this new version?

```
Request to node rabbit@xxxxx failed with {'EXIT',
{{badmatch,
{error,not_found}},
[{rabbit_queue_type,
discover,1,
[{file,
"rabbit_queue_type.erl"},
{line,250}]},
{rabbit_amqqueue,
augment_declare_args,
5,
[{file,
"rabbit_amqqueue.erl"},
{line,749}]},
{rabbit_channel,
handle_method,6,
[{file,
"rabbit_channel.erl"},
{line,2495}]}]}}
```

**Comment by lukebakken** [comment]
Please tell us *exactly* how you are creating the queue, and if you have set the default type.

Right now you're asking us to guess!

**Reply by ko2024** [reply]
Hi @lukebakken,

In console,  
Virtual host  /
Type: Default for virtual host

but if I select the classic queue it works normally

In console,  
Virtual host  /
Type: Classic

In version 3.11.10 it was possible to create in this format.

**Reply by ko2024** [reply]
had a scenario that in the new version 3.13.2, when not creating the default virtual host queue in the conventional way and selecting the classic type, the x-queue-type is inserted in the arguments after being created. And in this scenario there was a consumer who had to declare this argument in the consumer code

**Comment by lukebakken** [comment]
@ko2024 I can't reproduce what you report. Here is what I tried:

* Start RabbitMQ 3.13.2 using Docker
    ```
    docker run --publish 5672:5672 --publish 15672:15672 rabbitmq:3-management
    ```
* Visit the management UI page to show virtual host `/` properties
    ![image](https://github.com/rabbitmq/rabbitmq-server/assets/514926/52be0819-960c-4d42-aee6-23ef3d32536a)
* Create a queue using the default queue type
    ![image](https://github.com/rabbitmq/rabbitmq-server/assets/514926/b9959dda-fa75-4696-850d-b53fdafd5332)
* Check that the queue was created
    ![image](https://github.com/rabbitmq/rabbitmq-server/assets/514926/b8243c92-7f45-475a-81be-32eb4f29981d)

@ko2024 - question:

* Is your environment 100% fresh or an upgrade?
* If your environment is an upgrade, you MUST tell us what version of RabbitMQ you were using.

**Reply by ko2024** [reply]
Hi @lukebakken, the environment was created to migrate from 3.11.10 to 3.13.2 using the blue-green model

**Reply by lukebakken** [reply]
Could you please, very precisely, describe how you created your 3.13.2 environment? Did you import any definitions from 3.11.10 before trying to create these queues?

It's important that you provide _EVERY DETAIL_ that you can. I've already spent time on this issue that could have been more productive if I were given more information.

**Reply by ko2024** [reply]
I created the environment in 3.13.2, I exported definitions from version 3.11.10 and when trying to create a new queue this error occurs. Thank you in advance for your support

**Reply by michaelklishin** [reply]
@ko2024 so we are supposed to guess what was in those definitions, right?

**Reply by michaelklishin** [reply]
This is very likely an effect of https://github.com/rabbitmq/rabbitmq-server/issues/10515. See your definitions file to confirm that it is missing a default queue type. Or better yet, rename users, virtual hosts and so on and share the file.

Setting a default queue type for all nodes (https://github.com/rabbitmq/rabbitmq-server/pull/11163, shipped `3.13.3`) should help if that hypothesis is correct. This assumes that you want to use the same default in every virtual host.

**Comment by ko2024** [comment]
@michaelklishin ,
#11163 does make sense. thank you very much for the support
